### ISSUE TITLE
Dynamiclly load images/layers into RAM

### ISSUE THREAD

**Issue by softcat477** [DDMAL/Paco_classifier#28]
on [`5c7cd51`](https://github.com/DDMAL/Paco_classifier/tree/5c7cd5ea7b8480c0b79a56f3dd0b302e74b76f0d) the job will be killed by `oom_reaper` when loading images/layers that exceed the RAM limit. 

## Goal
Load part of images/layers that fit into the RAM limit. After doing gradient descent on those training data, kick some data out and load new images/pairs to RAM.
